### PR TITLE
Model loading

### PR DESCRIPTION
**Pull request by tigranfah**
* add llama 3.2 text models to YNNtitan
* our implementation is consistent with huggingface, the logits match between the two implementations
* add llama 3.2 configs

### CODE REVIEW

**Review by philippguevorguian — APPROVED** [review]
LGTM, ready to merge. 

For future reference on Rope implementation in case of rope implementation confusion:
- https://github.com/huggingface/transformers/issues/25199
- https://discuss.huggingface.co/t/is-llama-rotary-embedding-implementation-correct/44509/3

**Comment by tigranfah** [comment]
i know, hf loads query and value matrices differently, that is why we have permute and reverse_permute functions in the code to account for this.
